lumigator/python/mzai/backend/backend/api/routes/jobs.py

```diff
@@ -103,20 +105,57 @@ def list_jobs(
     )
 
 
-@router.get("/{job_id}")
-def get_job(service: JobServiceDep, job_id: UUID) -> Job:
+@router.get("/{job_spec}")
+def get_job(
+    service: JobServiceDep,
+    job_spec: str,
+    skip: int = 0,
+    limit: int = 100,
+) -> Job | ListingResponse[Job]:
     """Attempts to retrieve merged job data from the Lumigator repository and Ray.
+    If a valid UUID is provided, a single record is returned. If a valid
+    job type (currently `inference` or `evaluation`) is provided, then a list
+    of all jobs of the appropriate type is returned. Any other input results
+    in an error.
 
     NOTE: Lumigator repository data takes precedence over Ray metadata.
     """
-    job = service.get_job(job_id)
-    ray_job = _get_ray_job(job_id)
-
-    # Combine both types of response.
-    x = ray_job.model_dump()  # JobSubmissionResponse
-    y = job.model_dump()  # JobResponse
-    merged = {**x, **y}
-    return Job(**merged)
+    try:
+        job_id = UUID(job_spec)
+        job = service.get_job(job_id)
+        ray_job = _get_ray_job(job_id)
+
+        # Combine both types of response.
+        x = ray_job.model_dump()  # JobSubmissionResponse
+        y = job.model_dump()  # JobResponse
+        merged = {**x, **y}
+        return Job(**merged)
+    except ValueError:
+        # If job_spec is not a UUID, it's assumed to be a
```

Comment:
missing comment?

Reply:
Ouch, yep, it should go there :-/ But it'll go away if we make a separate endpoint, anyway.